src/test/rustdoc/issue-55364.rs

```diff
@@ -0,0 +1,69 @@
+// @has issue_55364/index.html
+// @has - '//a[@class="mod"]' 'subone'
+// @has issue_55364/subone/index.html
+// These foo/bar links should refer inside `subone`
+// @has - '//div[@class="docblock"]//a[@href="../../issue_55364/subone/fn.foo.html"]' 'foo'
```

Comment:
This check is a bit too generic. Elements such as functions/methods or anything **have** a class/id, please use it so we are sure the link is where it should.

Reply:
Oh also you might need `// ignore-tidy-linelength` at some point (it's limited to 100 chars and I think you'll overpass it).

Reply:
And to specify [`String::from_utf16`](https://doc.rust-lang.org/nightly/std/string/struct.String.html#method.from_utf16), you can use this xpath: `//h4[@id=method.from_utf16"]`. However I just realised that docblocks aren't embedded into the element (I should finish that experiment some day...).

Reply:
Ok, explanation about `@has`. The argument right after it (`-` in most cases) is the file (and `-` means "same as previously set). Please replace it when required. For example, for modules, please refer to their file precisely.

Reply:
Thanks for that, it was helpful.  I have now pushed a squashed commit where I've made the selectors as specific as I can.  I fear any more specific and they'll start to depend on the exact rendering of markdown which isn't really what we're testing here.